src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -103,17 +102,31 @@ private static (List<Error>, TableOfContentsInputModel) LoadTocModel(JToken tocT
                 throw Errors.CircularReference(file, parents).ToException();
             }
 
-            parents.Add(file);
+            var (errors, model) = LoadTocModel(context, file, content);
 
-            var (errors, models) = LoadTocModel(context, file, content);
+            TableOfContentsMetadata tocMetadata = null;
+            if (parents.Count == 0 && metadataProvider != null && monikerProvider != null)
```

Comment:
>parents.Count == 0  [](start = 16, length = 19)

fileMonikers == null?

Reply:
that's means the included TOC is using the root TOC's fileMonikers?

---
In reply to: [243783684](https://github.com/dotnet/docfx/pull/3797#discussion_r243783684) [](ancestors = 243783684)

Reply:
The nodes imported from included toc will be treated just like the node in the current toc file.
we just apply different rule  on the absolute link node/ relative link node/ text node.

---
In reply to: [243783774](https://github.com/dotnet/docfx/pull/3797#discussion_r243783774) [](ancestors = 243783774,243783684)